pydantic/fields.py

```diff
@@ -334,10 +340,14 @@ def _validate_sequence_like(
         """
         if not sequence_like(v):
             e: errors_.PydanticTypeError
-            if self.shape is Shape.LIST:
-                e = errors_.ListError()
-            elif self.shape is Shape.SET:
-                e = errors_.SetError()
+            # Map shapes to error types.
+            shapes_to_errors = {
```

Comment:
you can move this definition out of the function to the module level.

We need to check whether this impacts performance since this is in the critical path for pydantic.

Reply:
I think by default cython has the `use_switch` optimization enabled which means it will likely be much faster to keep this as a series of `if`s, even if uglier.

Let me know if you don't think that's the case.

Reply:
I agree with your assessment; a C switch statement should be faster than a Python dictionary lookup. To be honest, though, I made this change mostly because lint was yelling at me about the function being too complex if I simply added another `elif` clause. Do you have any suggestions on what to do here? Should I just disable the warning as is done above?

Reply:
fine just to add ignore as we do elsewhere.